pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -615,3 +615,35 @@ def is_module(self) -> bool:
         from pytorch_lightning import LightningDataModule, LightningModule  # prevent cyclic import
 
         return isinstance(self.instance, (LightningModule, LightningDataModule))
+
+
+@dataclass
+class _DataHookSource:
+    """Stores the info about the shared DataHooks within LightningModule and LightningDataModule.
+
+    The hook source can be
+
+    1. a method from the :class:`~pytorch_lightning.core.lightning.LightningModule`,
+    2. a method from the :class:`~pytorch_lightning.core.datamodule.LightningDataModule`,
+
+    Arguments:
+        model: A LightningModule
+        datamodule: A LightningDataModule
+    """
+
+    model: Optional["pl.LightningModule"]
+    datamodule: Optional["pl.LightningDataModule"]
+
+    def get_hook(self, hook_name):
```

Comment:
This class currently seems only to be used for  3 hook methods, then let's check if the hook_name is valid.